bodhi/tests/server/services/test_updates.py

```diff
@@ -5106,3 +5107,167 @@ def test_unexpected_exception(self, log_exception, waive_test_results, *args):
         self.assertEquals(res.json_body['errors'][0]['description'],
                           u'IOError. oops!')
         log_exception.assert_called_once_with("Unhandled exception in waive_test_results")
+
+
+class TestGetTestResults(BaseTestCase):
+    """
+    This class contains tests for the get_test_results() function.
+    """
+    @mock.patch.dict(config, [('greenwave_api_url', None)])
+    def test_cannot_get_test_results_when_no_greenwave_url(self, *args):
+        """
+        Ensure that we get an error if trying to get test results of an update
+        when there is no greenwave_api_url defined in the configuration.
+        """
+        nvr = u'bodhi-2.0-1.fc17'
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.locked = False
+
+        res = self.app.get('/updates/%s/get-test-results' % str(nvr), status=400)
```

Comment:
4xx codes indicate that the client made an error, and specifically 400 means the request is bad. In this case, the error is that the server isn't configured to talk to Greenwave. I suggest a 501 for not implemented.

Reply:
sure if you want...